### DISCUSSION TITLE
Gene Ontology Ribbon JSON formating

### DISCUSSION THREAD

**Discussion by ma3touch** [geneontology/helpdesk#477] [Q&A] [answered]
#### Hello geneontolodym,
hope you are well.
I want to ask a question in regards of using GO ribbon on JSON data, I'm trying to use it with a json formated data as you suggested in your example [here](https://geneontology.org/docs/ribbon.html) in the **Generic Ribbon, other usages and data model** section but when doing so I see that your component makes an API request to 
```API query is https://api.geneontology.org/api/ontology/ribbon/?subset=goslim_agr&subject=subject1&subject=subject2&subject=subject3&subject=subject4&subject=subject5&subject=subject6```,
you can check the attached sample that am using
[sample.json](https://github.com/geneontology/helpdesk/files/14407924/sample.json).
How to show case data without the need to that API call.
any information provided with be appreciated.
Thank you

**Comment by kltm** [comment]
@AyoubMaatouch Just to clarify a little here, what data are you looking for _exactly_ and/or what is the desired end result? Pretty much all of the data that is used in the ribbon can be found in an unprocessed form as part of our downloads (i.e. ontology [https://current.geneontology.org/ontology] and annotation [http://current.geneontology.org/annotations] data). The API is useful for some tasks, but is still very much a work in progress.

**Reply by ma3touch** [reply]
as I understood from the docs, I can use the ribbon on self owned data and that what am trying to do, I created a json file based on docs in [this page](https://geneontology.org/docs/ribbon.html) section **Generic Ribbon, other usages and data model**, but when doing so, the ribbon checks the subject ID and send an API call to your database, and that not what am trying to do. am trying to make it work with the data that I have. 
I hope that I made my idea a bit clear. 
Thanks again for your prompt response

**Comment by kltm** [comment]
@dustine32 @tmushayahama Given the clarification above, would you have any recommendations for somebody trying to run the ribbon in their own data?

**Comment by tmushayahama** [comment, marked as answer]
I believe you can change change the subjectUrl properties as per this documentations

https://github.com/geneontology/wc-ribbon/tree/master/wc-go-ribbon/src/components/go-ribbon

and also for reference, you can see how uniprot does it on https://github.com/ebi-uniprot/uniprot-website

tagging @kltm @AyoubMaatouch @dustine32

**Reply by ma3touch** [reply]
Thank you for your time 👍
